File: app/src/main/java/com/erlite/aprihive/methods/MyActionDialog.java
package com.erlite.aprihive.methods;

import android.app.Dialog;
import android.content.Context;
import android.util.Log;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.erlite.aprihive.R;




public class MyActionDialog {

    private final Dialog dialog;


    public MyActionDialog(Context context, String title, String msg, int image, Runnable action){

        dialog = new Dialog(context);
        dialog.setContentView(R.layout.custom_dialog);

        dialog.getWindow().setBackgroundDrawable(null);
        dialog.getWindow().setLayout(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT);
        dialog.setCancelable(false);

        TextView actionText =  dialog.findViewById(R.id.dialogText);
        TextView actionTitle = dialog.findViewById(R.id.dialogTitle);
        ImageView actionImage = dialog.findViewById(R.id.dialogImage);
        TextView positiveBtn = dialog.findViewById(R.id.positiveButton);
        TextView negativeBtn = dialog.findViewById(R.id.negativeButton);

        actionText.setText(msg);
        actionTitle.setText(title);
        actionImage.setImageResource(image);

        positiveBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                action.run();
                dialog.dismiss();

            }
        });
        negativeBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                dialog.dismiss();
            }
        });



    }

    public MyActionDialog(Context context, String title, String msg, int image, Runnable action, String btnText, int btnColor){

        dialog = new Dialog(context);
        dialog.setContentView(R.layout.custom_dialog);

        dialog.getWindow().setBackgroundDrawable(null);
        dialog.getWindow().setLayout(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT);
        dialog.setCancelable(true);

        TextView actionText =  dialog.findViewById(R.id.dialogText);
        TextView actionTitle = dialog.findViewById(R.id.dialogTitle);
        ImageView actionImage = dialog.findViewById(R.id.dialogImage);
        TextView positiveBtn = dialog.findViewById(R.id.positiveButton);
        TextView negativeBtn = dialog.findViewById(R.id.negativeButton);

        actionText.setText(msg);
        actionTitle.setText(title);
        actionImage.setImageResource(image);

        positiveBtn.setText(btnText);
        //positiveBtn.setBackgroundColor(btnColor);
        negativeBtn.setVisibility(View.GONE);

        positiveBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                action.run();
                dialog.dismiss();

            }
        });
        negativeBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                dialog.dismiss();
            }
        });



    }

    //get notification
    public MyActionDialog(Context context, String title, String msg, int image, int imageColor, Runnable action, String btnText, Boolean cancelable){

        dialog = new Dialog(context);
        dialog.setContentView(R.layout.custom_dialog);

        dialog.getWindow().setBackgroundDrawable(null);
        dialog.getWindow().setLayout(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT);
        dialog.setCancelable(cancelable);

        TextView actionText =  dialog.findViewById(R.id.dialogText);
        TextView actionTitle = dialog.findViewById(R.id.dialogTitle);
        ImageView actionImage = dialog.findViewById(R.id.dialogImage);
        TextView positiveBtn = dialog.findViewById(R.id.positiveButton);
        TextView negativeBtn = dialog.findViewById(R.id.negativeButton);

        actionText.setText(msg);
        actionTitle.setText(title);
        actionImage.setColorFilter(context.getResources().getColor(imageColor));
        actionImage.setImageResource(image);

        positiveBtn.setText(btnText);
        //positiveBtn.setBackgroundColor(btnColor);
        negativeBtn.setVisibility(View.GONE);

        positiveBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                action.run();
                dialog.dismiss();

            }
        });
        negativeBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                dialog.dismiss();
            }
        });



    }

    public MyActionDialog(Context context, String title, String msg, int image, Runnable action, String positiveBtnText, String negativeBtnText){

        dialog = new Dialog(context);
        dialog.setContentView(R.layout.custom_dialog);

        dialog.getWindow().setBackgroundDrawable(null);
        dialog.getWindow().setLayout(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT);
        dialog.setCancelable(true);

        TextView actionText =  dialog.findViewById(R.id.dialogText);
        TextView actionTitle = dialog.findViewById(R.id.dialogTitle);
        ImageView actionImage = dialog.findViewById(R.id.dialogImage);
        TextView positiveBtn = dialog.findViewById(R.id.positiveButton);
        TextView negativeBtn = dialog.findViewById(R.id.negativeButton);

        actionText.setText(msg);
        actionTitle.setText(title);
        actionImage.setImageResource(image);

        positiveBtn.setText(positiveBtnText);
        negativeBtn.setText(negativeBtnText);

        positiveBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                action.run();
                dialog.dismiss();

            }
        });
        negativeBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                dialog.dismiss();
            }
        });



    }

    public void show(){
        dialog.show();
    }

    public void dismiss(){
        dialog.dismiss();
    }


}

File: app/src/main/java/com/aprihive/methods/MyActionDialog.kt
package com.aprihive.methods

import android.app.Dialog
import com.aprihive.R
import android.view.ViewGroup
import android.widget.TextView
import com.google.android.material.snackbar.Snackbar
import com.aprihive.methods.MySnackBar
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.net.ConnectivityManager
import android.net.NetworkInfo
import android.os.Build
import androidx.appcompat.app.AppCompatDelegate
import android.content.SharedPreferences
import android.view.View
import android.widget.ImageView

class MyActionDialog {
    private val dialog: Dialog

    constructor(context: Context?, title: String?, msg: String?, image: Int, action: Runnable) {
        dialog = Dialog(context!!)
        dialog.setContentView(R.layout.custom_dialog)
        dialog.window!!.setBackgroundDrawable(null)
        dialog.window!!.setLayout(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT)
        dialog.setCancelable(false)
        val actionText = dialog.findViewById<TextView>(R.id.dialogText)
        val actionTitle = dialog.findViewById<TextView>(R.id.dialogTitle)
        val actionImage = dialog.findViewById<ImageView>(R.id.dialogImage)
        val positiveBtn = dialog.findViewById<TextView>(R.id.positiveButton)
        val negativeBtn = dialog.findViewById<TextView>(R.id.negativeButton)
        actionText.text = msg
        actionTitle.text = title
        actionImage.setImageResource(image)
        positiveBtn.setOnClickListener {
            action.run()
            dialog.dismiss()
        }
        negativeBtn.setOnClickListener { dialog.dismiss() }
    }

    constructor(context: Context, title: String?, msg: String?, image: Int, action: Runnable, btnText: String?, btnColor: Int) {
        dialog = Dialog(context)
        dialog.setContentView(R.layout.custom_dialog)
        dialog.window!!.setBackgroundDrawable(null)
        dialog.window!!.setLayout(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT)
        dialog.setCancelable(true)
        val actionText = dialog.findViewById<TextView>(R.id.dialogText)
        val actionTitle = dialog.findViewById<TextView>(R.id.dialogTitle)
        val actionImage = dialog.findViewById<ImageView>(R.id.dialogImage)
        val positiveBtn = dialog.findViewById<TextView>(R.id.positiveButton)
        val negativeBtn = dialog.findViewById<TextView>(R.id.negativeButton)
        actionText.text = msg
        actionTitle.text = title
        actionImage.setImageResource(image)
        positiveBtn.text = btnText
        positiveBtn.setBackgroundColor(context.resources.getColor(btnColor))
        negativeBtn.visibility = View.GONE
        positiveBtn.setOnClickListener {
            action.run()
            dialog.dismiss()
        }
        negativeBtn.setOnClickListener { dialog.dismiss() }
    }

    //get notification
    constructor(context: Context, title: String?, msg: String?, image: Int, imageColor: Int, action: Runnable, btnText: String?, cancelable: Boolean?) {
        dialog = Dialog(context)
        dialog.setContentView(R.layout.custom_dialog)
        dialog.window!!.setBackgroundDrawable(null)
        dialog.window!!.setLayout(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT)
        dialog.setCancelable(cancelable!!)
        val actionText = dialog.findViewById<TextView>(R.id.dialogText)
        val actionTitle = dialog.findViewById<TextView>(R.id.dialogTitle)
        val actionImage = dialog.findViewById<ImageView>(R.id.dialogImage)
        val positiveBtn = dialog.findViewById<TextView>(R.id.positiveButton)
        val negativeBtn = dialog.findViewById<TextView>(R.id.negativeButton)
        actionText.text = msg
        actionTitle.text = title
        actionImage.setColorFilter(context.resources.getColor(imageColor))
        actionImage.setImageResource(image)
        positiveBtn.text = btnText
        positiveBtn.setBackgroundColor(context.resources.getColor(R.color.color_text_blue_500))
        negativeBtn.visibility = View.GONE
        positiveBtn.setOnClickListener {
            action.run()
            dialog.dismiss()
        }
        negativeBtn.setOnClickListener { dialog.dismiss() }
    }

    constructor(context: Context?, title: String?, msg: String?, image: Int, action: Runnable, positiveBtnText: String?, negativeBtnText: String?) {
        dialog = Dialog(context!!)
        dialog.setContentView(R.layout.custom_dialog)
        dialog.window!!.setBackgroundDrawable(null)
        dialog.window!!.setLayout(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT)
        dialog.setCancelable(true)
        val actionText = dialog.findViewById<TextView>(R.id.dialogText)
        val actionTitle = dialog.findViewById<TextView>(R.id.dialogTitle)
        val actionImage = dialog.findViewById<ImageView>(R.id.dialogImage)
        val positiveBtn = dialog.findViewById<TextView>(R.id.positiveButton)
        val negativeBtn = dialog.findViewById<TextView>(R.id.negativeButton)
        actionText.text = msg
        actionTitle.text = title
        actionImage.setImageResource(image)
        positiveBtn.text = positiveBtnText
        negativeBtn.text = negativeBtnText
        positiveBtn.setOnClickListener {
            action.run()
            dialog.dismiss()
        }
        negativeBtn.setOnClickListener { dialog.dismiss() }
    }

    //with icon color
    constructor(context: Context, title: String?, msg: String?, image: Int, imageColor: Int, action: Runnable, positiveBtnText: String?, negativeBtnText: String?) {
        dialog = Dialog(context)
        dialog.setContentView(R.layout.custom_dialog)
        dialog.window!!.setBackgroundDrawable(null)
        dialog.window!!.setLayout(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT)
        dialog.setCancelable(true)
        val actionText = dialog.findViewById<TextView>(R.id.dialogText)
        val actionTitle = dialog.findViewById<TextView>(R.id.dialogTitle)
        val actionImage = dialog.findViewById<ImageView>(R.id.dialogImage)
        val positiveBtn = dialog.findViewById<TextView>(R.id.positiveButton)
        val negativeBtn = dialog.findViewById<TextView>(R.id.negativeButton)
        actionText.text = msg
        actionTitle.text = title
        actionImage.setImageResource(image)
        actionImage.setColorFilter(context.resources.getColor(imageColor))
        positiveBtn.text = positiveBtnText
        negativeBtn.text = negativeBtnText
        positiveBtn.setOnClickListener {
            action.run()
            dialog.dismiss()
        }
        negativeBtn.setOnClickListener { dialog.dismiss() }
    }

    fun show() {
        dialog.show()
    }

    fun dismiss() {
        dialog.dismiss()
    }
}